test/InterchainGovernance.js

```diff
@@ -151,39 +125,67 @@ describe('InterchainGovernance', () => {
             })
             .then((tx) => tx.wait());
 
-        await interchainGovernance.executeProposalAction(governanceChain, governanceAddress.address, payload);
+        await interchainGovernance.executeProposalAction(governanceChain, governanceAddress.address, payload).then((tx) => tx.wait());
 
-        await network.provider.send('evm_increaseTime', [timeDelay]);
-        await network.provider.send('evm_mine');
+        if (isHardhat) {
+            await network.provider.send('evm_increaseTime', [timeDelay]);
+            await network.provider.send('evm_mine');
+        } else {
+            await new Promise((resolve) => setTimeout(resolve, timeDelay * 1000));
+        }
 
         const tx = await interchainGovernance.executeProposal(target, calldata, nativeValue);
 
         await expect(tx).to.emit(interchainGovernance, 'ProposalExecuted').to.changeEtherBalance(target, nativeValue);
     });
 
-    it('should cancel an existing proposal', async () => {
+    it('should revert on calling withdraw directly', async () => {
+        const recipient = ownerWallet.address;
+        const nativeValue = 100;
+        await expect(interchainGovernance.withdraw(recipient, nativeValue)).to.be.revertedWithCustomError(interchainGovernance, 'NotSelf');
+    });
+
+    it('should withdraw native ether from governance to recipient', async () => {
         const commandID = 0;
-        const commandIDCancel = 1;
-        const target = targetContract.address;
+        const target = interchainGovernance.address;
         const nativeValue = 100;
-        const timeDelay = 12 * 60 * 60;
+        const recipient = ownerWallet.address;
 
-        const targetInterface = new Interface(['function callTarget() external']);
-        const calldata = targetInterface.encodeFunctionData('callTarget');
+        const withdrawInterface = new Interface(['function withdraw(address recipient, uint256 amount) external']);
```

Comment:
why hardcode the interface again? should create it dynamically
https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/test/Operators.js#L154

Reply:
updated